### PR TITLE
Moved from v1alpha1 to v1

### PR DESCRIPTION
**Pull request by jpkrohling**
Signed-off-by: Juraci Paixão Kröhling <juraci@kroehling.de>

### CODE REVIEW

**Comment by jpkrohling** [comment]
This change is [Reviewable](https://reviewable.io/reviews/jaegertracing/jaeger-operator/265)

**Comment by jpkrohling** [comment]
This PR is still missing the warning in case people try to use the old `v1alpha1`. This will be done in a subsequent commit to this PR, but it would be nice to get a first feedback on the changes so far.

**Comment by jpkrohling** [comment]
When a CR with the old `apiVersion` is deployed, this is what's shown in the operator's logs:

```
WARN[0000] Attention: you are using a legacy CR for Jaeger. Use the apiVersion 'jaegertracing.io/v1' instead of 'io.jaegertracing/v1alpha1'  instance=simplest namespace=default
```

File: test/e2e/sidecar.go [hunk 1]
```diff
@@ -161,6 +161,7 @@ func sidecarTest(t *testing.T, f *framework.Framework, ctx *framework.TestCtx) e
 		resp := &resp{}
 		err = json.Unmarshal(body, &resp)
 		if err != nil {
+			fmt.Println(body)
```
Comment (objectiser) [inline test/e2e/sidecar.go]: Did you mean to have this `Println`?

**Comment by jpkrohling** [comment]
> Just want to clarify - so the current PR will print a warning and not deploy anything, but before we release v1 it will handle deploying based on the old alpha1 version aswell?

We *could* do some type transformation. My preference would be to *not* have it for `v1alpha1` -> `v1`, only for `v1` -> `v1.next` and on.

File: CONTRIBUTING.adoc [hunk 2]
```diff
@@ -90,7 +90,7 @@ Similar instructions also work for OpenShift, but the target `run-openshift` can
 
 ==== Model changes
 
-The Operator SDK generates the `pkg/apis/io/v1alpha1/zz_generated.deepcopy.go` file via the command `make generate`. This should be executed whenever there's a model change (`pkg/apis/io/v1alpha1/types.go`)
+The Operator SDK generates the `pkg/apis/jaegertracing/v1/zz_generated.deepcopy.go` file via the command `make generate`. This should be executed whenever there's a model change (`pkg/apis/jaegertracing/v1/jaeger_types.go`)
```
[the remark this answers is not in the document]
Comment (jpkrohling) [reply CONTRIBUTING.adoc]: Once we migrate to the SDK 0.5.0, I'll change this, as there are more generated files.

File: deploy/crds/jaegertracing_v1_jaeger_cr.yaml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+apiVersion: jaegertracing.io/v1
```
[the remark this answers is not in the document]
Comment (jpkrohling) [reply deploy/crds/jaegertracing_v1_jaeger_cr.yaml]: That's where people expect the example to be. The "scorecard" also checks for a CR here.

File: deploy/crds/jaegertracing_v1_jaeger_crd.yaml [hunk 4]
```diff
@@ -0,0 +1,33 @@
+apiVersion: apiextensions.k8s.io/v1beta1
```
[the remark this answers is not in the document]
Comment (jpkrohling) [reply deploy/crds/jaegertracing_v1_jaeger_crd.yaml]: Users would have the CRD already, but if we need it for some test (like, when checking if we have a warning message if an old CR is installed), then we need this CRD installed in the cluster.

File: deploy/crds/jaegertracing_v1_jaeger_crd.yaml [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  group: jaegertracing.io
+  names:
+    kind: Jaeger
+    listKind: JaegerList
+    plural: jaegers
+    singular: jaeger
+  scope: Namespaced
+  subresources:
+    status: {}
+  validation:
```
[the remark this answers is not in the document]
Comment (jpkrohling) [reply deploy/crds/jaegertracing_v1_jaeger_crd.yaml]: I'll remove it, this comes from the SDK 0.5.0, which I used as reference for the new API.

**Comment by jpkrohling** [comment]
> Do we need to keep apis.go for v1aplha1 if we don't do model transhormation?

Yes, otherwise users will get the following error when trying to create a Jaeger instance with the old `apiGroup`:

```
error: unable to recognize "/tmp/simplest.yaml": no matches for kind "Jaeger" in version "io.jaegertracing/v1"
```

> It seems that some files reference io.jaegertracing. Like the OLM catalog

I fixed the OLM catalog, which will still include the old `apiGroup` for the time being. I fixed the one in the contribution guide as well. All other entries look fine to me. Could you please re-check?

**Comment by pavolloffay** [comment]
> I fixed the OLM catalog, which will still include the old apiGroup for the time being. 

We should maybe create an issue to migrate it to the new one?

**Comment by jpkrohling** [comment]
> We should maybe create an issue to migrate it to the new one?

I meant, it has both `apiGroups` now.

**Pull request merged**